app/src/main/java/com/github/se/polyfit/ui/components/AddIngredientPopup.kt

```diff
@@ -0,0 +1,197 @@
+              onValueChange = { newValue ->
+                nutritionSize[index] = removeLeadingZerosAndNonDigits(newValue)
+              },
+              modifier =
+                  Modifier.testTag("NutritionSizeInput " + nutritionLabels[index]).weight(0.5f),
+              singleLine = true,
+              colors =
+                  TextFieldDefaults.colors(
+                      focusedIndicatorColor = PrimaryPurple,
+                      unfocusedIndicatorColor = SecondaryGrey,
+                      focusedContainerColor = Color.Transparent,
+                      unfocusedContainerColor = Color.Transparent))
+
+          Text(
+              text = nutritionUnit[index],
+              style = TextStyle(fontSize = 18.sp),
+              color = SecondaryGrey,
+              modifier =
+                  Modifier.testTag("NutritionUnit " + nutritionLabels[index])
+                      .weight(0.4f)
+                      .padding(start = 8.dp))
+        }
+    Spacer(modifier = Modifier.height(10.dp))
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchIngredients() {
+  //  val searchText = remember { mutableStateOf("") }
+  var searchText by remember { mutableStateOf("") }
+
+  val showIngredientSearch = remember { mutableStateOf(false) }
+
+  // TODO: change list of ingredients to search for here
+  val ingredientDatabase = TMP_AVAILABLE_INGREDIENT
+
+  // TODO: (Optional) filter ingredients here or at backend
+  val filteredIngredients =
+      remember(searchText) {
+        ingredientDatabase.filter { it.contains(searchText, ignoreCase = true) }
+      }
+
+  SearchBar(
+      modifier = Modifier.padding(start = 20.dp, end = 20.dp).testTag("SearchIngredientBar"),
+      query = searchText,
+      onQueryChange = { searchText = it },
+      onSearch = { Log.v("Search", "Searching for $searchText") },
+      active = showIngredientSearch.value,
+      onActiveChange = { showIngredientSearch.value = true },
+      leadingIcon = {
+        Icon(Icons.Filled.Search, contentDescription = "search", tint = SecondaryGrey)
+      },
+      placeholder = {
+        Text(
+            text = "Find an Ingredient...",
+            color = SecondaryGrey,
+            style = TextStyle(fontSize = 17.sp))
+      }) {
+        LazyColumn(modifier = Modifier.testTag("IngredientSearchScrollableList")) {
+          items(filteredIngredients) { ingredient ->
+            Text(
+                text = ingredient,
+                modifier =
+                    Modifier.testTag("SearchResult $ingredient")
+                        .fillMaxWidth()
+                        .clickable {
+                          searchText = ingredient
+                          showIngredientSearch.value = false
+                          // TODO: Apply according nutrition facts to nutrition text fields
+                        }
+                        .padding(8.dp))
+            Divider()
+          }
+        }
+      }
+}
+
+@Composable
```

Comment:
Amazing-looking button! To enhance its reusability, it could be a good idea to store it in its own file. This way, we can easily reuse it to maintain a more consistent interface across the app.

Reply:
done